### ISSUE TITLE
Now dev dynamic routes doesn't work with Next when refreshing page 

### ISSUE THREAD

**Issue by SarKurd** [vercel/vercel#2351]
It does work after pushing it to Zeit

![k4L2UaIcYU](https://user-images.githubusercontent.com/13942717/57817093-26ad7f80-777e-11e9-8ac9-9d6ffd712143.gif)

**now.json**
```
{
  "version": 2,
  "builds": [
    { "src": "next.config.js", "use": "@now/next" },
    { "src": "api/*.js", "use": "@now/node" }
  ],
  "routes": [
    {
      "src": "/player/(?<name>[^/]+)$",
      "dest": "/player?name=$name"
    }
  ]
}
```

**pages/index.js**

```javascript
import Link from "next/link";

export default () => (
  <div>
    <p>
      Search for
      <Link href="/player?name=xxsarkurdzz" as="/player/xxsarkurdzz">
        <a>xXSARKURDZz</a>
      </Link>
      !
    </p>
  </div>
);
```

**pages/player.js**
```javascript
const player = ({ name }) => (
  <div>
    <h1>{name}</h1>
    <p>Welcome to our player page for {name}!</p>
  </div>
);

player.getInitialProps = async ({ query }) => {
  return { name: query.name };
};

export default player;

```

**Comment by TooTallNate** [comment]
Can you try with `now@canary`? This was recently fixed in #2330, but is not yet on a stable release.

**Comment by SarKurd** [comment]
It works, thanks.
